packages/server/src/api/notifRoutes.ts

```diff
@@ -0,0 +1,112 @@
+import { ServerRoute, ResponseObject } from "@hapi/hapi";
+import { NotifBody } from "@template/common";
+import { NotifModel } from "../entity/NotifModel";
+import { NotifPayload } from "../joi";
+import * as dotenv from "dotenv";
+import * as webPush from "web-push";
+import { DeepPartial } from "typeorm";
+
+// configure env vars for VAPID
+dotenv.config();
+
+// if env vars not found, then throw an error early
+if (!process.env.EMAIL || !process.env.PUBLICKEY || !process.env.PRIVATEKEY) {
+  throw new Error(
+    "please add a .env file with keys+email in packages/server. ask alex/eddy for deets."
+  );
+} else {
+  webPush.setVapidDetails(
+    process.env.EMAIL,
+    process.env.PUBLICKEY,
+    process.env.PRIVATEKEY
+  );
+}
+
+export const notifRoutes: ServerRoute[] = [
+  {
+    // endpoint to get the VAPID public key
+    method: "GET",
+    path: "/api/v1/notifications/credentials",
+    handler: (request, h) => {
+      return JSON.stringify(process.env.PUBLICKEY);
+    },
+  },
+  {
+    method: "POST",
+    path: "/api/v1/notifications/register/{user_id}", // TODO:   make this not a param for the users lmaoooooooo soI don't spam alex with sugondese
+    handler: async (request, h): Promise<string | ResponseObject> => {
+      const payload = request.payload as NotifBody;
+      console.debug("registering user with endpoint:", payload.endpoint);
+      await NotifModel.create(
+        toDBmodel(payload, Number(request.params.user_id))
+      ).save();
+      return h.response(JSON.stringify("registration success")).code(200);
+    },
+    options: {
+      validate: {
+        payload: NotifPayload.options({ presence: "required" }),
+      },
+    },
+  },
+  {
+    method: "POST",
+    path: "/api/v1/notifications/notify_user/{user_id}",
```

Comment:
In prod, will we actually have this endpoint?

Reply:
short of checking that the user the TA is attempting to notify is in the queue and that the TA is helping them, probably... but I agree there probably are precautions that we should take to prevent a TA from using our API to mass spam everyone

Reply:
Yeah... maybe it should be a route on `/queues/:queue_id/questions/:question_id/notify` so that the notification is associated to the question ticket, not a catch all notify-anyone route.

Reply:
let's open an issue for this